dokuwiki-syntax/src/main/java/org/xwiki/contrib/dokuwiki/syntax/internal/parser/DokuWikiIterativeParser.java

```diff
@@ -897,6 +897,88 @@ private void processImage(
             if (imageArgument.startsWith(TAG_RSS_GREATER_THAN_SYMBOL)) {
                 break;
             }
+
+            if (imageArgument.startsWith(("youtube>"))) {
+                buffer.clear();
+                while (source.ready()) {
+                    c = source.read();
+
+                    if (c == '}') {
+                        if (!param.containsKey("url")) {
+                            param.put("url", "http://www.youtube.com/watch?v=" + getStringRepresentation(buffer));
+                            source.read();
+                            break;
+                        } else {
+                            param = setVideoSize(param, buffer, source);
+                        }
+                        break;
+                    }
+                    if (c == '?') {
+                        param.put("url", "http://www.youtube.com/watch?v=" + getStringRepresentation(buffer));
+                        buffer.clear();
+                        continue;
+                    }
+                    buffer.add((char) c);
+                }
+                listener.onMacro("video", param, null, true);
+                buffer.clear();
+                break;
+            }
+
+            if (imageArgument.startsWith(("dailymotion>"))) {
+                buffer.clear();
+                while (source.ready()) {
+                    c = source.read();
+
+                    if (c == '}') {
+                        if (!param.containsKey("url")) {
+                            param.put("url", "http://www.dailymotion.com/video/" + getStringRepresentation(buffer));
+                            source.read();
+                            break;
+                        } else {
+                            param = setVideoSize(param, buffer, source);
+                        }
+                        break;
+                    }
+                    if (c == '?') {
+                        param.put("url", "http://www.dailymotion.com/video/" + getStringRepresentation(buffer));
+                        buffer.clear();
+                        continue;
+                    }
+                    buffer.add((char) c);
+                }
+                listener.onMacro("video", param, null, true);
+                buffer.clear();
+                break;
+            }
+
+            if (imageArgument.startsWith(("vimeo>"))) {
```

Comment:
I'm not a fan of all this copy/pasting regarding macro which have almost exactly the same code.

Ideally the design should be more:
* a generic code find a dokuwiki plugin
* it then ask to ComponentManager if a DokuwikiPlugin component exist for the macro id ("youtube" for example)
* if one is found, call DokuwikiPlugin#handle (or some name like this) which will produce the right event for this macro
* if none is found calls onMacro with the id of the plugin and what can be seen as parameters in dokuwiki syntax